src/http-operation.ts

```diff
@@ -0,0 +1,199 @@
+  deprecated?: boolean;
+
+  // most of the type, params from oas2/oas3 will simply be converted to content with one '*' key
+  content: {
+    [mediaType: string]: IHttpOpContent;
+  };
+}
+
+export const enum HttpOpParamStyles {
+  Simple = 'simple',
+  Matrix = 'matrix',
+  Label = 'label',
+  Form = 'form',
+  SpaceDelimited = 'spaceDelimited',
+  PipeDelimited = 'pipeDelimited',
+  DeepObject = 'deepObject',
+}
+
+export interface IHttpOpPathParam extends IHttpOpParam {
+  // defaults to simple
+  style?: HttpOpParamStyles.Label | HttpOpParamStyles.Matrix | HttpOpParamStyles.Simple;
+}
+
+export interface IHttpOpQueryParam extends IHttpOpParam {
+  // defaults to form
+  style?:
+    | HttpOpParamStyles.Form
+    | HttpOpParamStyles.SpaceDelimited
+    | HttpOpParamStyles.PipeDelimited
+    | HttpOpParamStyles.DeepObject;
+
+  allowEmptyValue?: boolean;
+  allowReserved?: boolean;
+}
+
+export interface IHttpOpHeaderParam extends IHttpOpParam {
+  // defaults to simple
+  style?: HttpOpParamStyles.Simple;
+}
+
+export interface IHttpOpCookieParam extends IHttpOpParam {
+  // defaults to form
+  style?: HttpOpParamStyles.Form;
+}
+
+export interface IHttpOpEncoding {
+  property: string;
+  mediaType?: string;
+  headers?: IHttpOpHeaderParam[];
+  // deafults to form
+  style:
+    | HttpOpParamStyles.Form
+    | HttpOpParamStyles.SpaceDelimited
+    | HttpOpParamStyles.PipeDelimited
+    | HttpOpParamStyles.DeepObject;
+
+  explode?: boolean;
+  allowReserved?: boolean;
+}
+
+export interface IHttpOpContent {
+  mediaType: string;
+  schema?: ISchema;
+  examples?: INodeExample[];
+  encodings?: IHttpOpEncoding[];
+}
+
+// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#link-object
+export interface IHttpOpLink {
+  operationRef: string;
+  parameters: Array<{
+    name: string;
+    value?: any;
+    expression?: string;
+  }>;
+  requestBody?: any;
+  description?: string;
+  server?: IServer;
+}
+
+// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#requestBodyObject
+export interface IHttpOpRequestBody {
+  description?: string;
+  required?: boolean;
+  contents: IHttpOpContent[];
+}
+
+export interface IHttpOpRequestParams {
+  path?: IHttpOpPathParam[];
+  query?: IHttpOpQueryParam[];
+  headers?: IHttpOpHeaderParam[];
+  cookie?: IHttpOpCookieParam[];
+  body?: IHttpOpRequestBody;
+}
+
+export interface IHttpOpResponse {
+  // Note: code MAY contain uppercase "X" to indicate wildcard
+  // Examples: 200, 2XX, 4XX, XXX ("default" in OAS)
+  // When mocking, should select most specific defined code
+  code: string;
+  contents: IHttpOpContent[];
+  description?: string;
+  headers?: IHttpOpHeaderParam[];
+  links?: IHttpOpLink;
+}
+
+/**
+ * Security
+ */
+
+type HttpOpSecurityScheme =
+  | IApiKeySecurityScheme
+  | IBearerSecurityScheme
+  | IBasicSecurityScheme
+  | IOauth2SecurityScheme
+  | IOpenIdConnectSecurityScheme;
+
+export interface ISecurityScheme {
```

Comment:
Note to self: does this need to be exported?

Reply:
I don't think so, @chris-miaskowski?